images/ig-build/Dockerfile

```diff
@@ -1,16 +1,19 @@
-FROM openjdk:21-jdk-bullseye
+FROM openjdk:23-jdk-slim-bookworm
 LABEL maintainer="Josh Mandel"
 
-ENV NODE_VERSION=v18.18.2
-RUN apt-get update && \
-    apt-get -y install python3 python3-pip gosu openssl wget graphviz ruby2.7 ruby2.7-dev inotify-tools && \
+ENV NODE_VERSION=v20.13.1
+
+# Install dependencies, clean up, and set up non-root user
+RUN apt-get update && apt-get -y install curl git python3 python3-pip python3-venv python3-requests openssl wget graphviz ruby ruby-dev && \
     apt-get clean && \
-    pip3 install --upgrade  requests zulip && \
-    gem install jekyll jekyll-asciidoc && \
-    cd /tmp && \
-    wget --quiet https://nodejs.org/dist/${NODE_VERSION}/node-${NODE_VERSION}-linux-x64.tar.xz && \
-    cd /usr/local && \
-    tar --strip-components 1 -xf /tmp/node-${NODE_VERSION}-linux-x64.tar.xz
+    gem install jekyll jekyll-asciidoc
+
+# Download and install Node.js
+WORKDIR /tmp
+
+RUN wget --quiet https://nodejs.org/dist/${NODE_VERSION}/node-${NODE_VERSION}-linux-x64.tar.gz && \
+    tar --strip-components 1 -xf node-${NODE_VERSION}-linux-x64.tar.gz -C /usr/local && \
```

Comment:
Worth saying, The reason I'm not just apt installing node + npm here Is that they are older versions, The npm version brings in several critical cves that docker scout will complain about. That still might be the best route for now, since it will get better over. I think that would solve your cross platform issue?

Alternatively, wyour problem be solved with an additional build argument that you could supply?

Reply:
(See updated images in the next commit on this PR -- they should work for you. You'll have to build the base image with with `-t ghcr.io/fhir/ig-publisher-base` before building the "localdev" image.